packages/share/ios/Classes/SharePlugin.m

```diff
@@ -15,8 +15,30 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
 
   [shareChannel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
     if ([@"share" isEqualToString:call.method]) {
+      NSDictionary *arguments = [call arguments];
+
+      if ([arguments[@"text"] length] == 0) {
+        result([FlutterError errorWithCode:@"ARGUMENT_ERROR"
+                                   message:@"Non-empty text expected"
+                                   details:nil]);
+        return;
+      }
+
+      NSNumber *originX = arguments[@"originX"];
+      NSNumber *originY = arguments[@"originY"];
+      NSNumber *originWidth = arguments[@"originWidth"];
+      NSNumber *originHeight = arguments[@"originHeight"];
+
+      CGRect originRect;
+      if (originX != nil && originY != nil && originWidth != nil && originHeight != nil) {
+        originRect = CGRectMake([originX doubleValue], [originY doubleValue],
+                                [originWidth doubleValue], [originHeight doubleValue]);
+      }
+      NSLog(@"%@", NSStringFromCGRect(originRect));
+
       [self share:call.arguments
-          withController:[UIApplication sharedApplication].keyWindow.rootViewController];
+          withController:[UIApplication sharedApplication].keyWindow.rootViewController
+                atSource:originRect];
       result(nil);
     } else {
       result([FlutterError errorWithCode:@"UNKNOWN_METHOD"
```

Comment:
This must be `result(FlutterNotImplemented)`;

Reply:
Ah ok, done.